### ISSUE TITLE
Power Lines: Scaling Laws for Weight Decay and Batch Size in LLM Pre-training

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#391]
## Links
https://arxiv.org/abs/2505.13738

## Authors
[Shane Bergsma](https://arxiv.org/search/cs?searchtype=author&query=Bergsma,+S), [Nolan Dey](https://arxiv.org/search/cs?searchtype=author&query=Dey,+N), [Gurpreet Gosal](https://arxiv.org/search/cs?searchtype=author&query=Gosal,+G), [Gavia Gray](https://arxiv.org/search/cs?searchtype=author&query=Gray,+G), [Daria Soboleva](https://arxiv.org/search/cs?searchtype=author&query=Soboleva,+D), [Joel Hestness](https://arxiv.org/search/cs?searchtype=author&query=Hestness,+J)

## TLDR
本論文はLLM事前学習における重み減衰とバッチサイズのスケーリング則を提案する。AdamWのタイムスケール T_EMA がデータ量/モデルサイズ比(D/N)のべき乗則に、最適/臨界バッチサイズがデータ量(D)のべき乗則に従うことを実証した。これにより、計算コストと学習時間のトレードオフを考慮した効率的なハイパーパラメータ設定が可能になる。

## 1. 論文の概要
本研究は、大規模言語モデル（LLM）の事前学習における重要なハイパーパラメータである**重み減衰(λ)とバッチサイズ(B)について、モデルサイズ(N)やデータセットサイズ(D)に応じた最適な設定を予測するためのスケーリング則（Scaling Laws）**を体系的に調査・提案するものです。主要な発見として、以下の3点が挙げられます。
AdamWタイムスケールのスケーリング則: 重み減衰λの効果を捉える指標 T_EMA = B/(ηλD) が、トークン/パラメータ比（TPP = D/N）に対して正確なべき乗則に従うことを発見しました。これにより、任意のN, D, Bに対して最適な重み減衰 λ_opt を事前に予測できます。
バッチサイズのスケーリング則: 最小損失を達成する最適バッチサイズ (B_opt) と、データ並列性の効率が低下し始める臨界バッチサイズ (B_crit) が、どちらもモデルサイズNに依らず、データセットサイズDのべき乗則に従うことを明らかにしました。
Pareto最適解: これらの法則に基づき、学習時間と計算コストという2つの競合する目標のバランスを取るためのPareto最適な学習設定（N, D, Bの組み合わせ）を導出する方法論を提示しました。

## 2. 先行研究との違いと新規性
本研究は、以下の点で先行研究と一線を画し、新規性を持っています。
重み減衰(λ)の体系的なスケーリング: これまでの研究では学習率(η)やバッチサイズ(B)のスケーリングが主でしたが、本研究は重み減衰(λ)のスケーリング則を初めて体系的に定式化しました。T_EMAという統一的な指標を用いることで、λ, η, B, D の相互作用を明らかにしています。
バッチサイズのスケーリング要因の特定: 先行研究では、B_optやB_critは計算量(C)や損失(L)に依存するとされていましたが、本研究ではこれらがデータセットサイズ(D)に根本的に依存することを実証しました。これは、より本質的なスケーリングの駆動要因を特定した点で重要です。
実用的なB_crit推定手法: 一般的な学習率減衰スケジュール下でも適用可能な、新しいB_critの推定手法を提案しました。これにより、理論上だけでなく、実際の学習シナリオでB_critをより正確に見積もることが可能になります。

## 3. 貢献の実証方法
論文の主張は、広範かつ体系的な実験によって実証されています。
大規模な実験: 111Mから3.3Bパラメータまで、複数のモデルサイズ(N)と、20 TPPから1280 TPPまで幅広いデータセットサイズ(D)の組み合わせで、合計約400のLLM学習実験を実施しました。
μP(Maximal Update Parameterization)の活用: 全てのモデルでμPを適用し、モデルサイズに応じた学習率(η)のスケーリングを標準化しました。これにより、他のハイパーパラメータ（特にλとB）が性能に与える影響をより純粋に分離・分析することが可能になりました。
べき乗則のフィッティングと検証: 実験で得られた最適なハイパーパラメータのデータ点に対し、べき乗則をフィッティングし、非常に高い決定係数（R² > 0.94）を確認しました。さらに、フィッティングに使用していない大規模モデルのデータ点もその法則によく乗ることを示し（Figure 1の凡例参照）、提案した法則の汎化性能の高さを証明しました。

<img width="783" height="392" alt="Image" src="https://github.com/user-attachments/assets/10319cc1-30e4-4c1d-b885-8fe4018b81d2" />

<img width="777" height="351" alt="Image" src="https://github.com/user-attachments/assets/93a13028-d66c-4030-8435-a4167b1b1f1c" />

## 4. 重要なFigureに関する説明
Figure 1: Hyperparameters and their power lines (ハイパーパラメータとそのべき乗則)
この図は本論文の最も重要な発見を3つのプロットで示しています。
左図 (Optimal T_EMA): AdamWタイムスケール(T_EMA)が、トークン/パラメータ比(TPP)のべき乗則 T_EMA ∝ TPP^-0.527 に従って減少することを示します。これは、モデルに対してデータ量が多い（過学習気味の）設定ほど、λを大きくして過去の勾配情報を早く忘れるべきであることを意味します。
中図 (Optimal batch size): 最適バッチサイズ(B_opt)が、モデルサイズに関係なく、データセットサイズ(D)のべき乗則 B_opt ∝ D^0.383 に従って増加することを示しています。
右図 (Critical batch size): 臨界バッチサイズ(B_crit)も同様に、目標損失達成に必要な最小データ量(D_min)のべき乗則 B_crit ∝ D_min^0.462 に従うことを示しています。これらの明確な「べき乗則の線（Power Lines）」が、本論文のタイトルの由来です。

<img width="768" height="335" alt="Image" src="https://github.com/user-attachments/assets/32d712a4-3f63-427d-a187-d28da071d5ae" />

Figure 6: Iso-loss curves illustrating time/compute Pareto frontier (時間/計算量のParetoフロンティアを示す等損失曲線)
本研究の知見を応用し、学習時間と計算コストのトレードオフを可視化した図です。
特定の損失目標（例：L=2.6）を達成するための、異なる学習設定（N, D, Bの組み合わせ）における時間（横軸）と計算コスト（縦軸）の関係を示します。
この図は、単一の「最適」な設定は存在せず、時間と計算コストのどちらを優先するかによって最適な選択肢が変わることを示唆しています。例えば、時間を最優先する場合、計算コスト最適（20 TPP）な設定よりも、より小さいモデルをより多くのデータで過学習させた方が高速であるという、直感に反するが戦略的に重要な結論を導き出しています。

<img width="779" height="371" alt="Image" src="https://github.com/user-attachments/assets/3bf59e3a-7a5b-430d-8c60-bc220c5f9618" />
